release notes/v0.34.0.md

```diff
@@ -0,0 +1,61 @@
+k6 v0.34.0 is here! :tada: It introduces the already announced `k6/execution` API and includes some more enhancements and a bunch of minor bug fixes.
+
+## New Features
+
+* k6/execution API
```

Comment:
```suggestion
## New k6 JavaScript API - `k6/execution`
```

This seems both more readable ([check out how the current one is rendered](https://github.com/grafana/k6/blob/release-notes-34/release%20notes/v0.34.0.md)) and more informative